### PR TITLE
Fixes for Operate release process discovered in 8.5.0-test1

### PR DESCRIPTION
**Pull request by cmur2**
## Description

This PR collects the fixes that are needed to let the Operate release process complete successfully in non-dry-run mode in the monorepo.

Operate release dry-run test: https://github.com/camunda/zeebe/actions/runs/8372894013/job/22924810666 ✔️ 

Operate release test: https://github.com/camunda/zeebe/actions/runs/8373271118/job/22925990780 ✔️ 

## Related issues

Related to https://github.com/camunda/team-infrastructure/issues/616

### CODE REVIEW

File: .github/workflows/operate-release-manual.yml
Comment (cmur2) [inline .github/workflows/operate-release-manual.yml:61]: ⚠️ Remove before merge!

File: .github/workflows/operate-release-reusable.yml
Comment (houssain-barouni) [inline .github/workflows/operate-release-reusable.yml:179]: what is the effect of `generate_release_notes : true` ? what did you get when you did the test?

Comment (cmur2) [reply .github/workflows/operate-release-reusable.yml:179]: No verbose release notes (not sure why), GitHub instead inserted a link to the commit range between tag 8.4.5 and tag operate-8.5.0-test1 - I think it also got confused of the merge history.

Comment (cmur2) [reply .github/workflows/operate-release-reusable.yml:179]: I'd still keep the option since it doesn't hurt and the release manager can choose to overwrite.

Comment (cmur2) [reply .github/workflows/operate-release-reusable.yml:179]: API docs: https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release

**Review by houssain-barouni — APPROVED** [review]

**Pull request merged**
